.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,45 @@
+name: Benchmark
+
+on:
```

Comment:
Let's change to a cron job of once a day or 12 hours.

Reply:
I wonder if we can skip if it is the same commit on main at that time?

Reply:
yeah in fact ideally we can skip if no files in each respective crate have changed

Reply:
Okay I added a sort of hacky solution but should be good enough

Reply:
Nifty trick